examples/connect/dash/app.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# mypy: ignore-errors
+import os
+
+from posit.connect.external.databricks import viewer_credentials_provider
+
+from databricks import sql
+from databricks.sdk.service.iam import CurrentUserAPI
+from databricks.sdk.core import ApiClient, Config
+
+from dash import Dash, html, Output, Input, dash_table
+import pandas as pd
+import flask
+
+DATABRICKS_HOST=os.getenv("DATABRICKS_HOST")
+DATABRICKS_HOST_URL = f"https://{DATABRICKS_HOST}"
+SQL_HTTP_PATH = os.getenv("DATABRICKS_PATH")
+
+df = None
+app = Dash(__name__)
+
+app.layout = html.Div(children=[
+    html.Div(id="greeting", children="Loading..."),
+    html.Div(id='table-container'),
+    html.Div(id="dummy")  # dummy element to trigger callback on page load
+])
+
+@app.callback(
+    [
+        Output("table-container", "children"),
+        Output("greeting", "children")
+    ],
+    Input("dummy", "children")
+)
+
+def update_page(_):
+    """
+    Dash example application that shows user information and 
+    the first few rows from a table hosted in Databricks.
+    """
+
+    session_token = flask.request.headers.get('Posit-Connect-User-Session-Token')
+    credentials_provider = viewer_credentials_provider(user_session_token=session_token)
```

Comment:
I see a pattern like this in all of the examples, and I wonder if we can wrap this up more. Can we have a single function that detects, or try/excepts, to get the header from the places where shiny/flask/etc. put it? (Tangentially related: #54)

Reply:
I've had the same thought. I think that would be a nice addition. We'll need to follow the same pattern we did with `pandas` to avoid having our SDK depend on each of the frameworks.

```python
try:
  import flask
except ImportError:
  return None
```

It also varies quite a bit depending on the framework. For flask/fastapi, the headers are on the incoming request object. For shiny it's on the `session` which is only available in the server block. We'll have to experiment with each of the frameworks more to see what feels like the right interface for this helper.

Reply:
> For shiny it's on the session which is only available in the server block

Slighly complicated further by the fact that Shiny Express doesn't have a server block. But still, may be worth exploring.